spec/std/process_spec.cr

```diff
@@ -442,31 +442,75 @@ describe Process do
+    it { Process.parse_arguments_posix(%q(foo\ bar)).should eq(["foo bar"]) }
+    it { Process.parse_arguments_posix(%q("foo\ bar")).should eq(["foo\\ bar"]) }
+    it { Process.parse_arguments_posix(%q('foo\ bar')).should eq(["foo\\ bar"]) }
+    it { Process.parse_arguments_posix("\\").should eq(["\\"]) }
+    it { Process.parse_arguments_posix(%q["foo bar" '\hello/' Fizz\ Buzz]).should eq(["foo bar", "\\hello/", "Fizz Buzz"]) }
+    it { Process.parse_arguments_posix(%q[foo"bar"baz]).should eq(["foobarbaz"]) }
+    it { Process.parse_arguments_posix(%q[foo'bar'baz]).should eq(["foobarbaz"]) }
+    it { Process.parse_arguments_posix(%(this 'is a "'very wei"rd co"m"mand please" don't do t'h'a't p"leas"e)).should eq(["this", "is a \"very", "weird command please", "dont do that", "please"]) }
 
     it "raises an error when double quote is unclosed" do
       expect_raises ArgumentError, "Unmatched quote" do
-        Process.parse_arguments(%q["foo])
+        Process.parse_arguments_posix(%q["foo])
       end
     end
 
     it "raises an error if single quote is unclosed" do
       expect_raises ArgumentError, "Unmatched quote" do
-        Process.parse_arguments(%q['foo])
+        Process.parse_arguments_posix(%q['foo])
+      end
+    end
+  end
+
+  describe ".parse_arguments_windows" do
+    it { Process.parse_arguments_windows(%q[]).should eq(%w[]) }
+    it { Process.parse_arguments_windows(%q[ ]).should eq(%w[]) }
+    it { Process.parse_arguments_windows(%q[foo]).should eq(%w[foo]) }
+    it { Process.parse_arguments_windows(%q[foo bar]).should eq(%w[foo bar]) }
+    it { Process.parse_arguments_windows(%q["foo bar" 'foo bar' baz]).should eq(["foo bar", "'foo", "bar'", "baz"]) }
+    it { Process.parse_arguments_windows(%q["foo bar"baz]).should eq(["foo barbaz"]) }
+    it { Process.parse_arguments_windows(%q[foo"bar baz"]).should eq(["foobar baz"]) }
+    it { Process.parse_arguments_windows(%q[foo\bar]).should eq(["foo\\bar"]) }
+    it { Process.parse_arguments_windows(%q[foo\ bar]).should eq(["foo\\", "bar"]) }
+    it { Process.parse_arguments_windows(%q[foo\\bar]).should eq(["foo\\\\bar"]) }
+    it { Process.parse_arguments_windows(%q[foo\\\bar]).should eq(["foo\\\\\\bar"]) }
+    it { Process.parse_arguments_windows(%q[ /LIBPATH:C:\crystal\lib ]).should eq(["/LIBPATH:C:\\crystal\\lib"]) }
+    it { Process.parse_arguments_windows(%q[a\\\b d"e f"g h]).should eq(["a\\\\\\b", "de fg", "h"]) }
+    it { Process.parse_arguments_windows(%q[a\\\"b c d]).should eq(["a\\\"b", "c", "d"]) }
+    it { Process.parse_arguments_windows(%q[a\\\\"b c" d e]).should eq(["a\\\\b c", "d", "e"]) }
+    it { Process.parse_arguments_windows(%q["foo bar" '\hello/' Fizz\ Buzz]).should eq(["foo bar", "'\\hello/'", "Fizz\\", "Buzz"]) }
+    it { Process.parse_arguments_windows(%q[this 'is a "'very wei"rd co"m"mand please" don't do t'h'a't p"leas"e"]).should eq(["this", "'is", "a", "'very weird", "command", "please don't do t'h'a't please"]) }
```

Comment:
These examples are a bit hard to read when using different string literal styles for the input and output values.
For example `%q[foo\\bar]` and `"foo\\\\bar"` looks like the number of backslashes doubles, but it's actually just extra escapes for the literal.
I'd rather use `%q` literals consistently.

Reply:
I think they are acceptable because the point of these specs is not to show that the output is literally a partition of the input. Do you prefer a plain array of `%q` literals in these specs?

Reply:
Yes, I would prefer that. Or use double quoted strings for the input (that would be more noise, but at least comparable).
My point is that you should be able to have a clear comparison between input and output (and see that its not just `split`) without the mental barrier of different literal formats.